src/groupeddataframe/groupeddataframe.jl

```diff
@@ -767,6 +766,10 @@ const GroupKeyTypes = Union{GroupKey, Tuple, NamedTuple, AbstractDict{Symbol}, A
 # All allowed scalar index types
 const GroupIndexTypes = Union{Integer, GroupKeyTypes}
 
+# GroupedDataFrame is not a multidimensional array, so it does not support cartesian indexing
```

Comment:
we had an inconsistency that sometimes we errored when `CartesianIndex` was passed and sometimes we worked. Now we fail always, which is expected as `GroupedDataFrame` is not an array.